components/async-container/async-container.js

```diff
@@ -0,0 +1,43 @@
+import '../loading-spinner/loading-spinner.js';
+import { AsyncContainerMixin, asyncStates } from './async-container-mixin.js';
+import { html, LitElement } from 'lit-element/lit-element.js';
+
+class AsyncContainer extends AsyncContainerMixin(LitElement) {
+
+	static get properties() {
+		return {
+			_showPending: { type: Boolean }
+		};
+	}
+
+	constructor() {
+		super();
+		this._showPending = false;
+	}
+
+	render() {
+		if (this.asyncState === asyncStates.success) {
+			return html`<slot></slot>`;
+		} else if (this.asyncState === asyncStates.failure) {
+			return html`<slot name="failure"></slot><slot></slot>`;
+		} else if (this.asyncState === asyncStates.pending && this._showPending) {
+			return html`<d2l-loading-spinner></d2l-loading-spinner>`;
+		} else {
+			return html`<slot name="initial"></slot>`;
+		}
+	}
+
+	updated(changedProperties) {
+		if (!changedProperties.has('asyncState')) return;
+		if (this.asyncState === asyncStates.pending && changedProperties.get('asyncState') === asyncStates.initial) {
+			setTimeout(() => {
```

Comment:
Currently, this delay is baked into this component, but I am considering whether it would be better to build this into the mixin's resolution logic (`finally` statements), and perhaps expose an overridable `asyncPendingDelay` property.

Reply:
Right... so you'd set a value (default `0`) that tells it your tolerance for how long you'd like it before notifying you that things are actually pending. So if everything resolves quickly (in under 100ms say), it would just go from initial -> success. But if things take longer, then it would go into pending first.